kernel/syscall.c

```diff
@@ -5804,7 +5856,41 @@ static long _syscall(void* args_)
         case SYS_bpf:
             break;
         case SYS_execveat:
-            break;
+        {
+            int dirfd = (int)x1;
+            const char* filename = (const char*)x2;
+            char** argv = (char**)x3;
+            char** envp = (char**)x4;
+            int flags = (int)x5;
+
+            _strace(
+                n,
+                "dirfd=%d filename=%s argv=%p envp=%p flags=%d",
+                dirfd,
+                filename,
+                argv,
+                envp,
+                flags);
+
+            /* free the previous kernel stack from the SYS_execve syscall */
+            if (thread->exec_kstack)
+            {
+                myst_put_kstack(thread->exec_kstack);
+                thread->exec_kstack = NULL;
+            }
+
+            /* the kstack is freed later by next exec or by exit */
+            thread->exec_kstack = args->kstack;
```

Comment:
`SYS_execveat` and `SYS_execve` both handles kstacks the same way before and after the call to `myst_syscall_execveat`. Consider moving the handling code into `myst_syscall_execveat` to avoid duplication.